src/networkrequest.cpp

```diff
@@ -248,6 +264,14 @@ void NetworkRequest::processData(QNetworkReply::NetworkError error,
   m_finalStatusCode = status;
 #endif
 
+  // Check for expected HTTP reponse statuses. This can distinguish successful
+  // responses (eg: 200 vs. 201), but it allows for expected error conditions
+  // (eg: 404 when it's okay for the resource not to exist).
+  if ((status != 0) && m_expectedStatusCodes.contains(status)) {
+    emit requestCompleted(data);
+    return;
+  }
+
```

Comment:
Why keep line 283 once we've added this block? I'm not against it, just curious.

Reply:
The behaviour before this PR was that an `m_expectedStatusCode` of zero was a special value meaning "accept any HTTP status code" and we retain that behaviour when the `m_expectedStatusCodes` is empty. In such a case, this `if` statement will evaluate to `false` and we fall-through all the way down to the `emit requestCompleted()` down on line 283.